app/workers/publishing_api_document_republishing_worker.rb

```diff
@@ -12,25 +12,79 @@
 # and sending it again after republishing. This also changes the version
 # numbering and would probably appear in the version history.
 class PublishingApiDocumentRepublishingWorker < WorkerBase
+  attr_reader :published_edition, :pre_publication_edition
+
   def perform(document_id)
     document = Document.find(document_id)
-    published_edition = document.published_edition
-    pre_publication_edition = document.pre_publication_edition
-
-    if published_edition
-      Whitehall::PublishingApi.locales_for(published_edition).each do |locale|
-        # We need to do this synchronously because we need to guarantee it
-        # completes before pushing the pre_publication_edition.
-        PublishingApiWorker.new.perform('Edition', published_edition.id, 'republish', locale.to_s)
+    #this the latest edition in a visible state ie: withdrawn, published
+    @published_edition = document.published_edition
+    #this is the latest edition in a non visible state - draft, scheduled
+    #unpublished editions (other than withdrawn) will be in draft state with
+    #an associated unpublishing
+    @pre_publication_edition = document.pre_publication_edition
+
+    if the_document_has_been_unpublished?
+      send_draft_and_unpublish
+    else
+      if there_is_only_a_draft?
+        send_draft_edition
+      elsif there_is_only_a_published_edition?
+        send_published_edition
+      elsif there_is_a_newer_draft?
+        send_published_edition
+        send_draft_edition
       end
```

Comment:
If we think we've covered all eventualities, can we add a `else raise "Should never happen"` or something similar so that we get notified if we've missed something?